drivers/counter/counter_nrfx_timer.c

```diff
@@ -283,7 +283,14 @@ static uint32_t get_pending_int(const struct device *dev)
 static int init_timer(const struct device *dev,
 		      const struct counter_timer_config *config)
 {
-	const struct counter_nrfx_config *nrfx_config = dev->config;
+	struct counter_nrfx_config *nrfx_config = (struct counter_nrfx_config *) dev->config;
```

Comment:
cast should be conditioned to simulation, otherwise config is not mutable.

Overall I do not see any problems with these changes, but abusing config for potentially mutable data doesn't feel right, so I'd discuss this in the near future before it propagates to other drivers.